File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        
            int i=0;
            int j=0;
            int maxim=0;
            int flag=0;
            int flag_indx=-1;
            while(j<nums.size()){
                
                    if(nums[j]==0 and flag==0){
                            flag=1;
                            flag_indx=j;
                    }
                
                    else if(nums[j]==0 and flag==1){
                            
                            i=flag_indx+1;
                             flag_indx=j;
                    }
                
                    maxim=max(maxim,j-i+1);
                    j++;
                
            }
        
        
            return maxim;
    }
};

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        
        indx=-1
        start=0
        i=0
        maxLen=0
        while(i<len(nums)):
            
            if nums[i]==0 and indx==-1:
                    indx=i
                    
                    
            elif nums[i]==0 and indx!=-1:
                print(i,start)
                maxLen=max(maxLen,i-start)    
                start=indx+1
                indx=i
             
            i+=1
                
        
        maxLen=max(maxLen,i-start)
        return maxLen
                    
                
            
        
            